lib/utils/lazy.ts

```diff
@@ -0,0 +1,32 @@
+/*
+Copyright 2020 Balena
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import * as BalenaSdk from 'balena-sdk';
+
+// Equivalent of _.once but avoiding the need to import lodash for lazy deps
+const once = <T>(fn: () => T) => {
+	let cached: T;
+	return (): T => {
+		if (!cached) {
+			cached = fn();
+		}
+		return cached;
+	};
+};
+
+export const getBalenaSdk = once(() =>
+	(require('balena-sdk') as typeof BalenaSdk).fromSharedOptions(),
```

Comment:
This line could be written as:
```js
(require('balena-sdk') as typeof import('balena-sdk')).fromSharedOptions(),
```
... in which case the module-level import line could be deleted:

~~`import * as BalenaSdk from 'balena-sdk';`~~

I imagine that it was your intention for `lazy.ts` to be generic/reusable for other functions besides `getBalenaSdk` (which may not depend on `balena-sdk`), so it would be desirable to avoid the module-level import. I gather that the current module-level import is for Typescript type information only, and gets deleted from built javascript. And I've also seen your draft PR that replaces `import` with `import type` for clarity and certainty (nice!), though it seems to be blocked by required changes to `prettier` (?). Nonetheless, at least while we don't have `import type`, I think the intention and behaviour are made clearer by moving the import inside `getBalenaSdk` as suggested. Built javascript appears to be unchanged.

Reply:
I'd rather wait for the `import type` as I think the `import()` is a bit weird, and yeah the reason we're waiting on prettier is that it fails to parse the `import type` into AST so it can't prettify it and fails but I wouldn't expect it to take long to be resolved